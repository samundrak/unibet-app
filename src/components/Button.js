import React from 'react';
import styled from 'styled-components';

const Btn = styled.button`
  color: white;
  background: #59aa37;
  border-style: none;
  border-radius: 5px;
  padding: 10px;
`;
const Button = ({ children, ...rest }) => {
  return <Btn {...rest}>{children}</Btn>;
};
export default Button;
