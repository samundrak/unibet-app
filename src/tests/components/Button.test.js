import React from 'react';
import { render } from 'react-dom';
import { shallow } from 'enzyme';
import Button from '../../components/Button';

describe('Test button', () => {
  it('should render simple button  ', () => {
    const button = shallow(<Button>Primary Button</Button>);
    expect(button.contains('Primary Button'));
  });

  it('should call callback function when clicked', () => {
    const container = document.createElement('div');
    const mockFn = jest.fn();
    const wrapper = shallow(<Button onClick={mockFn}>Simple Button</Button>);
    wrapper.simulate('click');
    expect(mockFn).toHaveBeenCalledTimes(1);
    wrapper.simulate('click');
    expect(mockFn).toHaveBeenCalledTimes(2);
  });
  it('should test snapshot', () => {
    const container = document.createElement('div');
    render(<Button>Click me</Button>, container);
    expect(container.innerHTML).toMatchSnapshot();
  });
});
