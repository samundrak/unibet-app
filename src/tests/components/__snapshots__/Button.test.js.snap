// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test button should test snapshot 1`] = `"<button class=\\"sc-bdVaJa dOEnpf\\">Click me</button>"`;
